static/js/components/material/Dialog.js

```diff
@@ -35,13 +38,19 @@ export default class Dialog extends React.Component {
   }
 
   componentDidMount() {
-    const { open, onAccept, onCancel } = this.props;
+    const { open, onAccept, onCancel, hideDialog } = this.props;
 
     // Hack to get dialog to play nicely with JS tests
     if (!this.dialogRoot || !this.dialogRoot.dataset) return;
 
     this.dialog = new MDCDialog(this.dialogRoot);
     this.dialog.listen('MDCDialog:accept', onAccept);
+    // Hack: If the redux action to hide the dialog is executed as part of the dialog accept
+    // handler, the dialog object doesn't tear down properly, resulting in browser scrollbars
+    // remaining inaccessible. This is fixed by attaching a second click handler to hide the
+    // dialog after 'onAccept' completes.
+    // $FlowFixMe: Flow thinks this.dialog might be null
+    this.dialog.listen('MDCDialog:accept', hideDialog || onCancel);
```

Comment:
If onCancel is customized and does not call hideDialog  itself, the dialog won't open again after onCancel is called.  For example, I added this to the ShareDialog:

```javascript
  customCancel = (): void => {
    console.log('CANCELLED');
  };
...
    return <Dialog
      title="Share this Video"
      onCancel={this.customCancel}
```
and when the `Close` button is pressed `customCancel` is called, the dialog still closes, but can't be opened again.  This can be fixed by changing `customCancel` to call `hideDialog()`;
```javascript
  customCancel = (): void => {
    const { hideDialog } = this.props;
    console.log('CANCELLLED');
    hideDialog();
  };
```
but I think maybe the more foolproof way is to add `hideDialog` as a second click handler to the MDC `cancel` event too:
```javascript
    if (hideDialog) {
      // $FlowFixMe: Flow thinks this.dialog might be null
      this.dialog.listen('MDCDialog:cancel', hideDialog);
    }
    // $FlowFixMe: Flow thinks this.dialog might be null
    this.dialog.listen('MDCDialog:cancel', onCancel);
```